irohad/network/block_loader.hpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_BLOCK_LOADER_HPP
+#define IROHA_BLOCK_LOADER_HPP
+
+#include "dao/dao.hpp"
+#include "rxcpp/rx-observable.hpp"
+
+namespace iroha {
+  namespace network {
+
+    /**
+     * Interface for downloading blocks from a network
+     */
+    class BlockLoaderApi {
+     public:
+      /**
+       * Method requests missed blocks from external peer starting from it's top
+       * block.
+       * Note, that blocks will be in order: from the newest
+       * to your actual top block.
+       * This order is required for verify blocks before storing in a ledger.
+       * @param target_peer - peer for requesting blocks
+       * @param topBlock - your last actual block
+       * @return observable with blocks
+       */
+      virtual rxcpp::observable <dao::Block> requestBlocks(
+          dao::Peer &target_peer, dao::Block &topBlock) = 0;
```

Comment:
I think we don't need to use `target_peer`.
It is enough to `std::string target_ip`.